modules/project/shared-vpc.tf

```diff
@@ -59,7 +67,7 @@ resource "google_compute_shared_vpc_service_project" "service_projects" {
 
 resource "google_compute_shared_vpc_service_project" "shared_vpc_service" {
   provider        = google-beta
-  count           = local.svpc_service_config.host_project != null ? 1 : 0
+  count           = var.shared_vpc_service_config.host_project != null ? 1 : 0
```

Comment:
One more thing. Can you add a precondition here to ensure that `grant_iam_for_services` only contains APIs mentioned in the yaml?

Reply:
One of the usage patterns that I would like to allow is something like that:
```hcl
module "service-project" {
  source          = "./fabric/modules/project"
  services = [
    "alloydb.googleapis.com",
    "run.googleapis.com",
  ]
  shared_vpc_service_config = {
    host_project = module.project.project_id
    grant_iam_for_services = module.service-project.services
  }
}
```

I know that I'm contradicting here a bit myself from the above comment, but ability not to repeat myself, and not having to check in YAML file, which services are allowed here, is reducing overhead when using this module.

Let's allow user to have magic in their own code 🧙